app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -144,11 +150,38 @@ class ExplorationActivityPresenter @Inject constructor(
     return ephemeralStateResult.getOrDefault(Exploration.getDefaultInstance())
   }
 
-  private fun backPressActivitySelector(backflowScreen: Int?){
-    when(backflowScreen){
-      ParentActivityForExploration.BACKFLOW_SCREEN_STORY.value -> activity.startActivity(StoryActivity.createStoryActivityIntent(context, internalProfileId, topicId, storyId))
-      ParentActivityForExploration.BACKFLOW_SCREEN_LESSONS.value -> activity.startActivity(TopicActivity.createTopicPlayStoryActivityIntent(activity, internalProfileId, topicId, storyId))
-      else -> activity.startActivity(TopicActivity.createTopicActivityIntent(context, internalProfileId, topicId))
+  private fun backPressActivitySelector(backflowScreen: Int) {
+    when (backflowScreen) {
+      BackflowScreenEnum.BACKFLOW_SCREEN_STORY.value -> activity.startActivity(
+        StoryActivity.createStoryActivityIntent(
+          context,
+          internalProfileId,
+          topicId,
+          storyId
+        )
+      )
+      BackflowScreenEnum.BACKFLOW_SCREEN_LESSONS.value -> activity.startActivity(
+        TopicActivity.createTopicPlayStoryActivityIntent(
+          activity,
+          internalProfileId,
+          topicId,
+          storyId
+        )
+      )
+      BackflowScreenEnum.BACKFLOW_SCREEN_DEFAULT.value -> activity.startActivity(
+        TopicActivity.createTopicActivityIntent(
+          context,
+          internalProfileId,
+          topicId
+        )
+      )
+      else -> activity.startActivity(
```

Comment:
Is there a need to have an else statement here when we are already handling everything through the default value ?